src/main/java/com/sprint/mission/discodeit/service/basic/BasicUserService.java

```diff
@@ -107,8 +126,42 @@ public void userIsExist(UUID id) {
 
     // 유저 삭제 (회원 탈퇴)
     // 불러온 데이터 List로 변환
-    private List<User> changeList(Map<UUID, User> map) {
+    private List<User> changeMapToList(Map<UUID, User> map) {
 
         return map.values().stream().toList();
     }
+
+    private void validationUser(User user) {
+        Email email = user.getEmail();
+        String name = user.getName();
+
+        if (checkIsEmailExist(email)) {
+            throw new IllegalArgumentException("이미 가입된 이메일입니다.");
+        } else if (checkIsNameExist(name)) {
+            throw new IllegalArgumentException("이미 존재하는 이름입니다.");
+        }
+    }
+
+    // 이메일 가입 여부 확인
+    private boolean checkIsEmailExist(Email email) {
+
+        String emailString = email.toString();
+
+        // userRepository에서 가져온 Map에서 email만 추출하여 List 형태로 저장
+        List<String> emails = userRepository.load().values().stream()
+                .map(user -> user.getEmail().toString())
+                .toList();
+
+        return emails.contains(email.toString());    // 가입된 이메일일 경우 true 반환
```

Comment:
이렇게 하면 코드가 더 간결해질 것 같아요.  
```suggestion
        return userRepository.load().values().stream()
                .map(user -> user.getEmail().toString())
                .anyMatch(emailString::equals);
```